src/main/java/org/dependencytrack/resources/v1/AnalysisResource.java

```diff
@@ -166,8 +169,13 @@ public Response updateAnalysis(AnalysisRequest request) {
             }
 
             String commenter = null;
-            if (getPrincipal() instanceof LdapUser || getPrincipal() instanceof ManagedUser || getPrincipal() instanceof OidcUser) {
-                commenter = ((UserPrincipal) getPrincipal()).getUsername();
+            if (getPrincipal() instanceof UserPrincipal principal) {
+                commenter = principal.getUsername();
+            } else if (getPrincipal() instanceof ApiKey apiKey) {
+                List<Team> teams = apiKey.getTeams();
+                List<String> teamNames = new ArrayList<String>();
+                teams.forEach(team -> teamNames.add(team.getName()));
+                commenter = String.join(", ", teamNames);
```

Comment:
Some trivia: Even though the schema and code make it look like an API key can be assigned to multiple teams, in reality that will never be the case. It's a bit weird. 

That being said, the way you implemented this will work just fine either way.

Reply:
Thought that to, but implemented it either way. So should I remove it, so there will be no more misunderstandings?